dbrequests/mysql/connection.py

```diff
@@ -44,9 +44,15 @@ def _send_data_update(self, df, table, mode='replace', **params):
             self._insert_update(df, table, tmp_table)
 
     def _write_csv(self, df, file):
-        # We have to convert any obj64 types to str64: Frame.to_csv can't
-        # handle obj64 types on it's own.
-        df = df[:, f[:].remove(f[obj64]).extend(str64(f[obj64]))][:, df.names]
+        # Before writing, we need to convert all columns to strings for two
+        # reasons:
+        # - We have to convert any obj64 types to str64: Frame.to_csv can't
+        #   process them.
+        # - We have to replace None with NULL to tell MySQL, that we have
+        #   actual NULL values. An empty cell is sometimes, but not always a
+        #   NULL value. See #30
+        df = df[:, f[:].remove(f[:]).extend(str64(f[:]))][:, df.names]
+        df.replace(None, 'NULL')
```

Comment:
Shouldnt it be converted to \N ? and not String Null as described in the SO post? (https://stackoverflow.com/questions/2675323/mysql-load-null-values-from-csv-data/2675493#2675493)  I tried it myself and wasnt able to get it to work because it was always escaping. using r"\N" was one way to actually write \N in the string but on trying to write to DB it was transferred into "\\N"

Reply:
We have disabled escape sequence interpretation for reading in data: this was the fix for #28 . We sometimes have strings containing '\\' and for them to be treated literally, we had to switch that off:https://github.com/INWTlab/dbrequests/blob/17febe1ea920e1f8c6972072a10f11c4fb7a4b7d/dbrequests/mysql/connection.py#L60 That means, however, that `\N` is not interpreted as NULL. In our current scenario NULL is still interpreted as NULL.